### PR TITLE
add EM-based translator for Ubiquity Journals

### PR DESCRIPTION
**Pull request by adam3smith**
Increasingly common provider of hosting for open access journals. Tests are for the three different schemas in the target regex

### CODE REVIEW

**Comment by adam3smith** [comment]
closes #948

File: Ubiquity Journals.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
+{
+	"translatorID": "5c11e3bd-caf5-4da6-95d8-e67c57929098",
+	"label": "Ubiquity Journals",
+	"creator": "Sebastian Karcher",
+	"target": "/articles/10\\.\\d{4,9}/[-._;()/:a-z0-9A-Z]+|/articles/?$|/\\d+/volume/\\d+/issue/\\d+",
```
Comment (zuphilip) [inline Ubiquity Journals.js]: Should we force here also that there is no other substructere before, i.e. something like
```
"target": "^[^/]+(/articles/10\\.\\d{4,9}/[-._;()/:a-z0-9A-Z]+|/articles/?$|/\\d+/volume/\\d+/issue/\\d+)",
```
?

File: Ubiquity Journals.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+				"ISSN": "2050-5833",
+				"abstractNote": "<p class=\"p1\">This paper explores the relation between the ‘DIY’ (‘do-it-yourself’) movement and ‘DIY architecture’, and the notion of social transformation, in examples of DIY manuals and discourse of North America drawn from the 1940s to the 1970s. The DIY movement emerged as a significant phenomenon in North America of the 1950s, where it was associated with a mainstream audience and a residential market. By the 1960s, the DIY approach was embraced by the North American counterculture as a self-sustaining sensibility that could overcome a reliance on the mainstream, consumerist society that spurned it. On the surface, the association of DIY with the counterculture and countercultural architects appears to denote a significant ideological shift from its original association with the beliefs and culture of mainstream North America and the nuclear family. However, one of the key characterisations of the DIY movement identified in the present paper is the way it is bound to the notion of social identity and transformation, regardless of ideology. Particular attention is paid to DIY manuals and discourse of the 1950s.</p>",
```
Comment (zuphilip) [inline Ubiquity Journals.js]: Maybe clean the `abstractNote` from the tags.

Comment (adam3smith) [inline Ubiquity Journals.js]: so - - there are actually escaped html tags (think &amp;lt;p&amp;gt;) in the abstract. That seemed to be so clearly just bogus on their part that I didn't bother cleaning them, but I guess I could? Will standard removeTags work?

File: Ubiquity Journals.js (hunk 1 above)
Comment (adam3smith) [inline Ubiquity Journals.js]: Let me check some more URLs, but I really like that as a general idea.

File: Ubiquity Journals.js (hunk 2 above)
Comment (zuphilip) [inline Ubiquity Journals.js]: I don't think `cleanTags` will work with escaped html tags, cf. https://github.com/zotero/zotero/blob/master/chrome/content/zotero/xpcom/utilities.js#L265 .  However, it should be easy to come up with something similiar, maybe only this case with the `<p>`-Tag. E.g. something like
```
item.abstractNote = abstract.replace(/&lt;p[^&>]*&gt;/g, "").trim();
```

**Comment by adam3smith** [comment]
used both of your suggestions; thanks @zuphilip

**Pull request merged**